File: scripts/append_questions.js
const fs = require('fs');
const path = require('path');

const filePath = process.argv[2];
const batchPath = process.argv[3];

if (!filePath || !batchPath) {
    console.error('Usage: node append_questions.js <file_path> <batch_path>');
    process.exit(1);
}

try {
    const existingData = JSON.parse(fs.readFileSync(filePath, 'utf8'));
    const newData = JSON.parse(fs.readFileSync(batchPath, 'utf8'));

    existingData.push(...newData);

    fs.writeFileSync(filePath, JSON.stringify(existingData, null, 4), 'utf8');
    console.log(`Successfully added ${newData.length} questions to ${filePath}`);
} catch (err) {
    console.error('Error:', err.message);
    process.exit(1);
}

File: scripts/append_questions.py
import json
import os
import sys

def append_questions(file_path, new_questions):
    if not os.path.exists(file_path):
        data = []
    else:
        with open(file_path, 'r', encoding='utf-8') as f:
            data = json.load(f)
    
    data.extend(new_questions)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=4)
    print(f"Successfully added {len(new_questions)} questions to {file_path}")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: python append_questions.py <file_path> <questions_json_file>")
        sys.exit(1)
    
    file_path = sys.argv[1]
    questions_file = sys.argv[2]
    
    with open(questions_file, 'r', encoding='utf-8') as f:
        new_q = json.load(f)
    
    append_questions(file_path, new_q)
